hoodie-common/src/main/java/com/uber/hoodie/common/util/CompactionUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.util;
+
+import com.uber.hoodie.avro.model.HoodieCompactionOperation;
+import com.uber.hoodie.avro.model.HoodieCompactionWorkload;
+import com.uber.hoodie.common.model.CompactionOperation;
+import com.uber.hoodie.common.model.FileSlice;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javafx.util.Pair;
+
+/**
+ * Helper class to generate compaction workload from FileGroup/FileSlice abstraction
+ */
+public class CompactionUtils {
```

Comment:
Should this live in hoodie-common? compaction is strictly a write operation correct?

Reply:
@vinothchandar : HoodieTableFileSystemView (in hoodie-common) in  dependent PRs uses this Utils class to track fileIds with pending compactions.    Hence, I had defined this class in hoodie-common